### PR TITLE
fix: manually disconnect peer connection before repairing new edge device

### LINKED ISSUE #710: In some cases UI shows edge device is offline while JS console shows otherwise
**Describe the bug**
`Timeline.vue` and `Settings.vue` create a new EdgeAPI object one `create()` which wraps around the current Peer object. However if the Peer object gets destroyed or it connects to a different remote Peer ID the `EdgeAPI` instance referenced in a vue page is not updated. That leads to a weird state where the new Peer object instance successfully ping/pongs with the connected edge device, but the UI shows disconnected state based on reference to an outdated instance.

We probably need to pull into Vuex all interrelated variables and ensure that they are in sync. When `peer` is recycled, `edgeAPI` should be recycled as well. 

We also need to add test coverage for these edge cases.

**To Reproduce**
1. Connect to an edge device.
2. Go to Settings page and plug-in a valid but offline remote edge id then try to connect to Remote device.
3. Go to Timeline and observe messages in JS console.
4. Go to Settings and observe messages to JS console.
5. Pair with local Ambianic Edge device. 
6. 3. Go to Timeline and observe messages in JS console.
7. Go to Settings and observe messages to JS console.

**Expected behavior**
UI pages and JS console should be in sync and display traffic to a single peer instance with a single active edge device.

**Screenshots**
See attached screenshots.

**Desktop (please complete the following information):**
 - OS: Mac
 - Browser: Chrome Version 92.0.4515.131 (Official Build) (x86_64)
 - Version: UI app version 2.17.0, Edge Software version: 1.15.5

Screen Shot 2021-08-24 at 10 55 13 AM

Screen Shot 2021-08-24 at 10 55 19 AM

Screen Shot 2021-08-24 at 11 04 07 AM

Screen Shot 2021-08-24 at 11 04 36 AM

Screen Shot 2021-08-24 at 10 47 45 AM

### PR DESCRIPTION
**Pull request by vickywane**
## Purpose
This pull request will close [this issue](https://github.com/ambianic/ambianic-ui/issues/710).

## Approach
This pull request modifies the `PEER_DISCONNECTED` mutation to manually disconnect the existing peer connection by calling the disconnect method on the peer instance before a new connection is established. 

To confirm, the `peer("disconnected")` event is triggered and `pnp client: Connection lost. Please reconnect.` is logged to the console.

## Merge Checklist
- [x]  The code change is tested and works locally.
- [x]  The user and dev documentation is up to date.
- [x]  There is no commented out code in this PR.
- [x]  No lint errors (use flake8)
- [x]  100% test coverage

### CODE REVIEW

**Review by ivelin — CHANGES_REQUESTED** [review]
Please make sure all checks pass before requesting a review.

File: src/store/pnp.js [hunk 1]
```diff
@@ -92,9 +92,15 @@ const mutations = {
     state.peer = peer
   },
   [PEER_DISCONNECTED] (state) {
-    state.peerConnection = undefined
-    state.peerConnectionStatus = PEER_DISCONNECTED
+    try {
+      state.peer.disconnect()
+    } catch (e) {
+      console.log(`Error disconnecting peer: ${e}`)
```
Comment (ivelin) [inline src/store/pnp.js:98]: @vickywane let's log this at error level instead of debug. `console.error`. We need to understand when `disconnect()` fails and fix it.

File: src/store/pnp.js [hunk 2]
```diff
@@ -531,8 +537,8 @@ const actions = {
    * them or let them connect to you.
    */
   async [CHANGE_REMOTE_PEER_ID] ({ state, commit, dispatch }, remotePeerId) {
-    commit(NEW_REMOTE_PEER_ID, remotePeerId)
     commit(PEER_DISCONNECTED)
```
Comment (ivelin) [inline src/store/pnp.js:540]: @vickywane good catch.

Comment (ivelin) [reply src/store/pnp.js:540]: @vickywane what UI flow was compromised due to this issue? Can you add it as a test case?

**Review by ivelin — CHANGES_REQUESTED** [review]
@vickywane Good observations. I'm glad you now understands better the p2p state machine so you can help solidify it.

A couple of related questions:

1. Do you think we should remove the continuous auto-discovery logic and make it more explicitly. Initiate local or remote discovery only when the user pushes a button instead of trying to figure out when it makes sense to auto-discover. I keep running into situations where the UX is in some weird state because there is a background auto-discovery running all the time.
2. The more I work on the p2p state machine, the less I like the name pnp. It meant something when I first wrote it, but it doesn't seem to make much sense now as the UI code expanded. Should we use a better name for the `pnp` module? `p2p` or `remote` maybe?

**Review by ivelin — CHANGES_REQUESTED** [review]
@vickywane please see comments on the PeerJS state machine and adjust the code logic accordingly.
